### PR TITLE
322 updgrade writable entity

### PR DESCRIPTION
**Pull request by LucasTesnier**
Add missing lock in writable clicking.

Dismissed the double taping functionality.

Add the enter keybind to unselect a writable.

Update the clicking functionality to avoid double click.

### CODE REVIEW

File: Client/ButtonAction.cpp [hunk 1]
```diff
@@ -73,6 +74,16 @@ void selectAWritable(World &world, Entity &entityPtr)
         return;
     }
     resource.window.setKeyRepeatEnabled(true);
+    auto joined = world.joinEntities<Selected, WritableContent>();
+    if (!joined.empty()) {
+        auto guardOld = std::lock_guard(*(joined.at(0)));
+        joined.at(0)->removeComponent<Selected>();
+        if (joined.at(0)->contains<TextureName>()) {
+            joined.at(0)->removeComponent<TextureName>();
+            joined.at(0)->addComponent<TextureName>(GraphicsTextureResource::WRITABLE);
```
Comment (31Nathan) [inline Client/ButtonAction.cpp:83]: Instead of removing and adding, just change the value

File: libs/GraphicECS/SFML/Systems/InputManagement.cpp [hunk 2]
```diff
@@ -82,6 +83,14 @@ namespace graphicECS::SFML::Systems
                 if (event.text.unicode == 8 && writableContent.content.size() != 0) {
                     auto guard = std::lock_guard(*entityPtr.get());
                     writableContent.content.pop_back();
+                } else if (event.text.unicode == 13) {
+                    auto guard = std::lock_guard(*entityPtr.get());
+                    entityPtr->removeComponent<Selected>();
+                    if (entityPtr->contains<TextureName>()) {
+                        entityPtr->removeComponent<TextureName>();
```
Comment (31Nathan) [inline libs/GraphicECS/SFML/Systems/InputManagement.cpp:90]: Same

**Review by 31Nathan — APPROVED** [review]

**Review by 31Nathan — CHANGES_REQUESTED** [review]
See the messages

File: Client/ButtonAction.cpp (hunk 1 above)
Comment (LucasTesnier) [reply Client/ButtonAction.cpp:83]: :eyes:

File: libs/GraphicECS/SFML/Systems/InputManagement.cpp (hunk 2 above)
Comment (LucasTesnier) [reply libs/GraphicECS/SFML/Systems/InputManagement.cpp]: :eyes:

**Review by 31Nathan — APPROVED** [review]

**Pull request merged**